Mathlib/Combinatorics/SimpleGraph/Extremal/TuranDensity.lean

```diff
@@ -0,0 +1,127 @@
+/-
+Copyright (c) 2025 Mitchell Horner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mitchell Horner
+-/
+import Mathlib.Analysis.Asymptotics.AsymptoticEquivalent
+import Mathlib.Combinatorics.SimpleGraph.DeleteEdges
+import Mathlib.Combinatorics.SimpleGraph.Extremal.Basic
+import Mathlib.Data.Nat.Choose.Cast
+
+/-!
+# Turán density
+
+This files defines the **Turán density** of a simple graph.
+
+## Main definitions
+
+* `SimpleGraph.turanDensity H` is the **Turán density** of the simple graph `H`, defined as the
+  limit of `extremalNumber n H / n.choose 2` as `n` approaches `∞`.
+
+* `SimpleGraph.tendsto_turanDensity` is the proof that `SimpleGraph.turanDensity` is well-defined.
+
+* `SimpleGraph.isEquivalent_extremalNumber` is the proof that `extremalNumber n H` is
+  asymptotically equivalent to `turanDensity H * n.choose 2` as `n` approaches `∞`.
+-/
+
+
+open Asymptotics Filter Finset Fintype Topology
+
+namespace SimpleGraph
+
+variable {V W : Type*} {G : SimpleGraph V} {H : SimpleGraph W}
+
+/-- If `G` is `H.Free`, then `G.deleteIncidenceSet v` is also `H.Free` and has at most
+`extremalNumber (card V-1) H` many edges. -/
+theorem card_edgeFinset_deleteIncidenceSet_le_extremalNumber
```

Comment:
I think this lemma might belong in a different file, perhaps the one defining the extremal number